BD-2476 Create SPM and Cocoapods section for Rich Push (#6281)
braze-inc/braze-docs: _docs/_developer_guide/platform_integration_guides/swift/push_notifications/customization/rich_notifications.md

The notification service extension's bundle ID must be distinct from your main app target's bundle ID. For example, if your app's bundle ID is `com.company.appname`, you can use `com.company.appname.AppNameServiceExtension` for your service extension.

## Step 3: Integrating rich push notifications

For a step-by-step guide on integrating rich push notifications with `BrazeNotificationService`, refer to our [tutorial](https://braze-inc.github.io/braze-swift-sdk/tutorials/braze/b2-rich-push-notifications).

To see a sample, refer to the usage in [`NotificationService`][1] of our Examples app.

### Adding the rich push framework to your app

{% tabs local %}
{% tab Swift Package Manager %}

After following the [Swift Package Manager integration guide]({{site.baseurl}}/developer_guide/platform_integration_guides/swift/initial_sdk_setup/installation_methods/swift_package_manager/), add `BrazePushStory` to your `Notification Content Extension`:

![In Xcode, under frameworks and libraries, select the "+" icon to add a framework.]({% image_buster /assets/img/swift/push_story/spm1.png %})

> ![]({% image_buster /assets/img/swift/push_story/spm2.png %})

Reviewer: "Rich Push" (gifs, video, images in push notifications) and "Push Stories" (a picture carousel) are actually two completely independent features. You can have one or the other, or both together

These screenshots actually show how to integrate "Push Stories" (hence, the name `BrazePushStory` package), and doesn't have to do with Rich Push

Author: Oh, yeah. The content in the tabs haven't been updated from the "Push Stories" article. I'm wondering if you or somebody else could provide the screenshots and information necessary to integrate Rich Push.

{% endtab %}
{% tab CocoaPods %}

> Add the following line to your Podfile:

Reviewer: ```suggestion
Add the following to your Podfile:
```

```ruby

> target 'YourAppTarget' do
> pod 'BrazeKit'
> pod 'BrazeUI'
> pod 'BrazeLocation'
> end
> 
> target 'YourNotificationServiceExtensionTarget' do
> pod 'BrazeNotificationService'
> end
> 
> target 'YourNotificationContentExtensionTarget' do
> pod 'BrazePushStory'
> end

Reviewer: Since this is a "Rich Push" article, I don't think we need to explicitly tell them how to integrate a totally separate feature (Push Stories). I added a comment to this section below:
```suggestion
target 'YourAppTarget' do
  pod 'BrazeKit'
  pod 'BrazeUI'
  pod 'BrazeLocation'
end

target 'YourNotificationServiceExtensionTarget' do
  pod 'BrazeNotificationService'
end

# Only include this below if your project integrates Push Stories as well
target 'YourNotificationContentExtensionTarget' do
  pod 'BrazePushStory'
end
```

```
{% alert note %}

> It's not required to have both Rich Push and Push Stories implemented in the same app. For instructions to implement Push Stories, see the [documentation]({{site.baseurl}}/developer_guide/platform_integration_guides/swift/push_notifications/push_story/?tab=swift%20package%20manager).

Reviewer: I think as long as we imply that these are two separate features, we don't need to call it out explicitly. Maybe this is enough:
```suggestion
If you wish to implement the Push Stories feature, see the [documentation]({{site.baseurl}}/developer_guide/platform_integration_guides/swift/push_notifications/push_story/?tab=swift%20package%20manager).
```

![][26]{: style="max-width:90%"}

Ensure that **Embed In Application** is set to embed the extension in your application.

## Step 2: Setting up the notification service extension

A notification service extension is its own binary that is bundled with your app. It must be set up in the [Apple Developer Portal][27] with its own app ID and provisioning profile.

The notification service extension's bundle ID must be distinct from your main app target's bundle ID. For example, if your app's bundle ID is `com.company.appname`, you can use `com.company.appname.AppNameServiceExtension` for your service extension.

## Step 3: Integrating rich push notifications

For a step-by-step guide on integrating rich push notifications with `BrazeNotificationService`, refer to our [tutorial](https://braze-inc.github.io/braze-swift-sdk/tutorials/braze/b2-rich-push-notifications).

To see a sample, refer to the usage in [`NotificationService`][1] of our Examples app.

### Adding the rich push framework to your app

{% tabs local %}
{% tab Swift Package Manager %}

> After following the [Swift Package Manager integration guide]({{site.baseurl}}/developer_guide/platform_integration_guides/swift/initial_sdk_setup/installation_methods/swift_package_manager/), add `BrazePushStory` to your `Notification Content Extension`:
> 
> ![In Xcode, under frameworks and libraries, select the "+" icon to add a framework.]({% image_buster /assets/img/swift/push_story/spm1.png %})
> 
> ![]({% image_buster /assets/img/swift/push_story/spm2.png %})

Reviewer: ```suggestion
After following the [Swift Package Manager integration guide]({{site.baseurl}}/developer_guide/platform_integration_guides/swift/initial_sdk_setup/installation_methods/swift_package_manager/), add `BrazeNotificationService` to your `Notification Service Extension`:

![In Xcode, under frameworks and libraries, select the "+" icon to add a framework.]
```

And here are the screenshots you can use:

<img width="965" alt="Screenshot 2023-10-09 at 5 44 38 PM" src="https://github.com/braze-inc/braze-docs/assets/4797040/7bc4495e-df6c-4c1f-85c7-f8e4d2e82c0d">

<img width="1124" alt="Screenshot 2023-10-09 at 6 01 02 PM" src="https://github.com/braze-inc/braze-docs/assets/4797040/064f282f-c32a-4a89-b950-b68d30d2b45a">
